src/core/Modal/BaseModal/BaseModal.tsx

```diff
@@ -0,0 +1,50 @@
+import { Modal, Button } from '@material-ui/core';
+import React from 'react';
+import useStyles from './styles';
+
+interface BaseModalProps {
+  isOpen: boolean;
+  handleClose: () => void;
+  hasCloseBtn: boolean;
+  textAlign?: string;
+}
+
+const BaseModal: React.FC<BaseModalProps> = ({
+  children,
+  isOpen,
+  handleClose,
+  hasCloseBtn,
+  textAlign,
+}) => {
+  const styleProps = { textAlign };
+  const classes = useStyles(styleProps);
+
+  return (
+    <Modal
+      open={isOpen}
+      onClose={handleClose}
+      disableBackdropClick
+      disableEscapeKeyDown
+      aria-labelledby="simple-modal-title"
+      aria-describedby="simple-modal-description"
+      className={classes.BaseModalStyle}
+    >
+      <div className={classes.modalContainer}>
+        {hasCloseBtn && (
+          <div className={classes.modalContainerClose}>
```

Comment:
Can we skip the modal container logic and keep something like

Base Component: Modal
Contains one global modal style: classes.root

If hasCloseBtn -> Add Button at Top right
Component: Button
Class: classes.closeBtn

```suggestion
          <Modal
      open={isOpen}
      onClose={handleClose}
      disableBackdropClick
      disableEscapeKeyDown
      aria-labelledby="simple-modal-title"
      aria-describedby="simple-modal-description"
      className={classes.root}>
            {hasCloseBtn && (
                 <Button ... className={classes.btnClose}>&#x2715;</Button>
            )}
            {children}
      </Modal>
```

Something similar